examples/oft-solana-v2/programs/oft/src/instructions/quote_send.rs

```diff
@@ -0,0 +1,193 @@
+        let amount_sent_ld = get_pre_fee_amount_ld(token_mint, amount_received_ld)?;
+
+        // remove the oft fee from the amount_received_ld
+        let oft_fee_ld = oft_store.remove_dust(calculate_fee(
+            amount_received_ld,
+            oft_store.default_fee_bps,
+            fee_bps,
+        ));
+        amount_received_ld -= oft_fee_ld;
+        (amount_sent_ld, amount_received_ld, oft_fee_ld)
+    } else {
+        // if it is Native OFT, there is no transfer fee
+        let amount_sent_ld = oft_store.remove_dust(amount_ld);
+        let oft_fee_ld = oft_store.remove_dust(calculate_fee(
+            amount_sent_ld,
+            oft_store.default_fee_bps,
+            fee_bps,
+        ));
+        let amount_received_ld = amount_sent_ld - oft_fee_ld;
+        (amount_sent_ld, amount_received_ld, oft_fee_ld)
+    };
+    Ok((amount_sent_ld, amount_received_ld, oft_fee_ld))
+}
+
+fn calculate_fee(pre_fee_amount: u64, default_fee_bps: u16, fee_bps: Option<u16>) -> u64 {
+    let final_fee_bps = if let Some(bps) = fee_bps { bps as u128 } else { default_fee_bps as u128 };
+    if final_fee_bps == 0 || pre_fee_amount == 0 {
+        0
+    } else {
+        // pre_fee_amount * final_fee_bps / ONE_IN_BASIS_POINTS
+        let fee = (pre_fee_amount as u128) * final_fee_bps;
+        (fee / ONE_IN_BASIS_POINTS) as u64
+    }
+}
+
+pub fn get_post_fee_amount_ld(token_mint: &InterfaceAccount<Mint>, amount_ld: u64) -> Result<u64> {
+    let token_mint_info = token_mint.to_account_info();
+    let token_mint_data = token_mint_info.try_borrow_data()?;
+    let token_mint_ext = StateWithExtensions::<MintState>::unpack(&token_mint_data)?;
+    let post_amount_ld =
+        if let Ok(transfer_fee_config) = token_mint_ext.get_extension::<TransferFeeConfig>() {
+            transfer_fee_config
+                .get_epoch_fee(Clock::get()?.epoch)
+                .calculate_post_fee_amount(amount_ld)
+                .ok_or(ProgramError::InvalidArgument)?
+        } else {
+            amount_ld
+        };
+    Ok(post_amount_ld)
+}
+
+// Calculate the amount_sent_ld necessary to receive amount_received_ld
+// Does *not* de-dust any inputs or outputs.
+fn get_pre_fee_amount_ld(token_mint: &InterfaceAccount<Mint>, amount_ld: u64) -> Result<u64> {
+    let token_mint_info = token_mint.to_account_info();
+    let token_mint_data = token_mint_info.try_borrow_data()?;
+    let token_mint_ext = StateWithExtensions::<MintState>::unpack(&token_mint_data)?;
+    let pre_amount_ld =
+        if let Ok(transfer_fee) = token_mint_ext.get_extension::<TransferFeeConfig>() {
+            calculate_pre_fee_amount(transfer_fee.get_epoch_fee(Clock::get()?.epoch), amount_ld)
+                .ok_or(ProgramError::InvalidArgument)?
+        } else {
+            amount_ld
+        };
+    Ok(pre_amount_ld)
+}
+
+// DO NOT CHANGE THIS CODE!!!
+// bug reported on token2022: https://github.com/solana-labs/solana-program-library/pull/6704/files
```

Comment:
what exactly is this comment pointing to?

Reply:
Anything in `examples/oft-solana-v2/programs` is out of scope as it is directly copied from audited monorepo code.